src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,87 @@
+        StringBuilder dataToWrite = new StringBuilder();
+        readDataFromFile(fromFileName);
+        String[] finalArrayOfData = generateArrayWithCorrectData(listOfFilesData, new String[2]);
+        int[] finalSumOfEachElementOfArrayData =
+                generateSumForEachElementOfCorrectDataArray(
+                        listOfFilesData,
+                        finalArrayOfData,
+                        0,
+                        0);
+        dataToWrite.append(finalArrayOfData[1]).append(",")
+                .append(finalSumOfEachElementOfArrayData[1])
+                .append(System.lineSeparator())
+                .append(finalArrayOfData[0]).append(",")
+                .append(finalSumOfEachElementOfArrayData[0])
+                .append(System.lineSeparator())
+                .append("result").append(",")
+                .append(finalSumOfEachElementOfArrayData[1] - finalSumOfEachElementOfArrayData[0]);
+        writeDataToFile(toFileName, dataToWrite);
+    }
+
+    public void writeDataToFile(String fileName, StringBuilder dataToWrite) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            writer.write(dataToWrite.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't find a file: " + fileName, e);
+        }
+    }
+
+    public void readDataFromFile(String fileName) {
+        String line;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while ((line = reader.readLine()) != null) {
+                listOfFilesData.addAll(Arrays.asList(line.split(",")));
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file: " + fileName, e);
+        }
+    }
+
+    public String[] generateArrayWithCorrectData(List<String> data, String[] result) {
+        for (int i = 0; i < data.size(); i += 2) {
+            for (int j = i + 2; j < data.size(); j += 2) {
+                if (data.get(i).equals(data.get(j))) {
+                    result[0] = data.get(j);
+                }
+            }
+        }
+        for (int i = 0; i < data.size(); i += 2) {
+            if (!result[0].equals(data.get(i))) {
+                result[1] = data.get(i);
+            }
+        }
+        return result;
+    }
 
+    public int[] generateSumForEachElementOfCorrectDataArray(
+            List<String> listOfFilesData,
+            String[] finalArrayOfData,
+            int firstSum,
+            int secondSum) {
+        for (int i = 0; i < listOfFilesData.size(); i++) {
+            if (listOfFilesData.get(i).equals(finalArrayOfData[0])) {
+                firstSum += Integer.parseInt(listOfFilesData.get(i + 1));
+            }
+            if (listOfFilesData.get(i).equals(finalArrayOfData[1])) {
+                secondSum += Integer.parseInt(listOfFilesData.get(i + 1));
+            }
+        }
```

Comment:
basically this logic has to be in createReport() method

Reply:
As far as I understood clean code correctly, we have to divide the logic into small methods which are responsible for a single action, so I don't think that makes sense, also the length of a method should not exceed 15 lines

Reply:
generateSumForEachElementOfCorrectDataArray
what is "sum for each element"
what is "correct data array"
why in methods name you have only array, but as parameters you pass list, some array "finalArray"?, firstSum, secondSum - This doesn't make sense

Reply:
Imagine looking at this code 2 months later. Would you still understand?